celer/tests/test_mtl.py

```diff
@@ -199,5 +202,52 @@ def test_GroupLassoCV(sparse_X):
     check_estimator(clf)
 
 
+def test_weights_group_lasso():
+    # generate data
+    sparse_X = 1
+    n_samples, n_features = 30, 50
+    X, y = build_dataset(n_samples, n_features, sparse_X=sparse_X)
+
+    # generate weihgts
+    groups = 5
+    n_groups = n_features // groups
+    np.random.seed(0)
+    weights = np.abs(np.random.randn(n_groups))
+
+    tol = 1e-8
+    params = {'n_alphas': 10, 'tol': tol, 'verbose': 1}
+    augmented_weights = (weights[:, None] *
+                         np.ones((n_groups, groups))).reshape((-1, ))
+
+    # method 1
+    alphas1, coefs1, gaps1 = celer_path(
+        X, y, "grouplasso", groups=groups, weights=weights,
+        eps=1e-2, **params)
+    # method 2
+    alphas2, coefs2, gaps2 = celer_path(
+        X.multiply(1 / augmented_weights[None, :]), y, "grouplasso",
```

Comment:
The mistake may be here : you pass both the weights and the scaled matrix, so you weight twice. Compare with the Lasso test

Reply:
I cannot see what you mean. I scale the matrix only once. 
I followed the same implementation as ``test_weights`` in ``test_lasso.py``.

Reply:
I m still unable to fix this test
Everything seems right, yet the test always fails

Reply:
where do you see failure ? with which error msg ?

Reply:
The two methods with weights and scaling with weights don't give the same result.
So, I get an Assertion error as the following:

```bash
E       AssertionError: 
E       Not equal to tolerance rtol=0.001, atol=0
E       
E       Mismatch: 45%
E       Max absolute difference: 2.90401975
E       Max relative difference: 125.45373015
E        x: array([[ 0.      ,  0.      ,  0.      ,  0.      ,  0.008718,  0.189253,
E                0.355291,  0.621276,  1.078897,  1.430047],
E              [ 0.      ,  0.      ,  0.      ,  0.      ,  0.005401,  0.112003,...
E        y: array([[ 0.000000e+00,  0.000000e+00,  0.000000e+00,  0.000000e+00,
E                0.000000e+00,  0.000000e+00,  0.000000e+00,  0.000000e+00,
E                0.000000e+00,  0.000000e+00],...

celer\tests\test_mtl.py:231: AssertionError
```

Reply:
Did you recompile the cython code ? Run pip install -e .

Reply:
My bad!
Now all is good